### ISSUE TITLE
Learning is failing and I don't know what the cause is. 

### ISSUE THREAD

**Issue by leeyunhome** [ultralytics/yolov5#4599]
## ❔Question
Hello,

Learning is failing and I don't know what the cause is. With the same dataset, we succeeded in learning 500 epochs with img size 384.

Thank you.

## Additional context

YOLOv5 🚀 v5.0-14-g238583b torch 1.8.0 CUDA:0 (GeForce RTX 3080, 10016.75MB)

Namespace(adam=False, artifact_alias='latest', batch_size=16, bbox_interval=-1, bucket='', cache_images=False, cfg='./models/yolov5s.yaml', data='/home/aiserver/coding/GitHub/trainingdb/data.yaml', device='', entity=None, epochs=400, evolve=False, exist_ok=False, global_rank=-1, hyp='data/hyp.scratch.yaml', image_weights=False, img_size=[416, 416], label_smoothing=0.0, linear_lr=False, local_rank=-1, multi_scale=False, name='bangbum_result_416_400epoch', noautoanchor=False, nosave=False, notest=False, project='runs/train', quad=False, rect=False, resume=False, save_dir='runs/train/bangbum_result_416_400epoch4', save_period=-1, single_cls=False, sync_bn=False, total_batch_size=16, upload_dataset=False, weights='yolov5s.pt', workers=8, world_size=1)
tensorboard: Start with 'tensorboard --logdir runs/train', view at http://localhost:6006/
hyperparameters: lr0=0.01, lrf=0.2, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=0.05, cls=0.5, cls_pw=1.0, obj=1.0, obj_pw=1.0, iou_t=0.2, anchor_t=4.0, fl_gamma=0.0, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0
wandb: Currently logged in as: hodu (use `wandb login --relogin` to force relogin)
wandb: wandb version 0.12.1 is available!  To upgrade, please run:
wandb:  $ pip install wandb --upgrade
wandb: Tracking run with wandb version 0.10.27
wandb: Syncing run bangbum_result_416_400epoch4
wandb: ⭐ View project at https://wandb.ai/hodu/YOLOv5
wandb: 🚀 View run at https://wandb.ai/hodu/YOLOv5/runs/2tuppu08
wandb: Run data is saved locally in /home/aiserver/coding/GitHub/yolov5/wandb/run-20210830_083324-2tuppu08
wandb: Run `wandb offline` to turn off syncing.

Overriding model.yaml nc=80 with nc=7

                 from  n    params  module                                  arguments                     
  0                -1  1      3520  models.common.Focus                     [3, 32, 3]                    
  1                -1  1     18560  models.common.Conv                      [32, 64, 3, 2]                
  2                -1  1     18816  models.common.C3                        [64, 64, 1]                   
  3                -1  1     73984  models.common.Conv                      [64, 128, 3, 2]               
  4                -1  1    156928  models.common.C3                        [128, 128, 3]                 
  5                -1  1    295424  models.common.Conv                      [128, 256, 3, 2]              
  6                -1  1    625152  models.common.C3                        [256, 256, 3]                 
  7                -1  1   1180672  models.common.Conv                      [256, 512, 3, 2]              
  8                -1  1    656896  models.common.SPP                       [512, 512, [5, 9, 13]]        
  9                -1  1   1182720  models.common.C3                        [512, 512, 1, False]          
 10                -1  1    131584  models.common.Conv                      [512, 256, 1, 1]              
 11                -1  1         0  torch.nn.modules.upsampling.Upsample    [None, 2, 'nearest']          
 12           [-1, 6]  1         0  models.common.Concat                    [1]                           
 13                -1  1    361984  models.common.C3                        [512, 256, 1, False]          
 14                -1  1     33024  models.common.Conv                      [256, 128, 1, 1]              
 15                -1  1         0  torch.nn.modules.upsampling.Upsample    [None, 2, 'nearest']          
 16           [-1, 4]  1         0  models.common.Concat                    [1]                           
 17                -1  1     90880  models.common.C3                        [256, 128, 1, False]          
 18                -1  1    147712  models.common.Conv                      [128, 128, 3, 2]              
 19          [-1, 14]  1         0  models.common.Concat                    [1]                           
 20                -1  1    296448  models.common.C3                        [256, 256, 1, False]          
 21                -1  1    590336  models.common.Conv                      [256, 256, 3, 2]              
 22          [-1, 10]  1         0  models.common.Concat                    [1]                           
 23                -1  1   1182720  models.common.C3                        [512, 512, 1, False]          
 24      [17, 20, 23]  1     32364  models.yolo.Detect                      [7, [[10, 13, 16, 30, 33, 23], [30, 61, 62, 45, 59, 119], [116, 90, 156, 198, 373, 326]], [128, 256, 512]]
Model Summary: 283 layers, 7079724 parameters, 7079724 gradients, 16.4 GFLOPS

Transferred 354/362 items from yolov5s.pt
Scaled weight_decay = 0.0005
Optimizer groups: 62 .bias, 62 conv.weight, 59 other
train: Scanning '/home/aiserver/coding/GitHub/trainingdb/train.cache' images and labels... 53083 found, 6 missing, 70 empty, 153 corrupted: 100%|█████████████████████████████| 53089/53089 [00:00<?, ?it/s]
val: Scanning '/home/aiserver/coding/GitHub/trainingdb/val.cache' images and labels... 6241 found, 1 missing, 14 empty, 0 corrupted: 100%|██████████████████████████████████████| 6242/6242 [00:00<?, ?it/s]
Plotting labels... 
/home/aiserver/coding/GitHub/yolov5/utils/plots.py:288: UserWarning: 
This call to matplotlib.use() has no effect because the backend has already
been chosen; matplotlib.use() must be called *before* pylab, matplotlib.pyplot,
or matplotlib.backends is imported for the first time.

The backend was *originally* set to 'TkAgg' by the following code:
  File "train.py", line 24, in <module>
    import test  # import test.py to get mAP after each epoch
  File "/home/aiserver/coding/GitHub/yolov5/test.py", line 12, in <module>
    from models.experimental import attempt_load
  File "/home/aiserver/coding/GitHub/yolov5/models/experimental.py", line 7, in <module>
    from models.common import Conv, DWConv
  File "/home/aiserver/coding/GitHub/yolov5/models/common.py", line 15, in <module>
    from utils.datasets import letterbox
  File "/home/aiserver/coding/GitHub/yolov5/utils/datasets.py", line 23, in <module>
    from utils.general import check_requirements, xyxy2xywh, xywh2xyxy, xywhn2xyxy, xyn2xy, segment2box, segments2boxes, \
  File "/home/aiserver/coding/GitHub/yolov5/utils/general.py", line 22, in <module>
    from utils.metrics import fitness
  File "/home/aiserver/coding/GitHub/yolov5/utils/metrics.py", line 5, in <module>
    import matplotlib.pyplot as plt
  File "/home/aiserver/anaconda3/envs/lpr_env/lib/python3.7/site-packages/matplotlib/pyplot.py", line 71, in <module>
    from matplotlib.backends import pylab_setup
  File "/home/aiserver/anaconda3/envs/lpr_env/lib/python3.7/site-packages/matplotlib/backends/__init__.py", line 16, in <module>
    line for line in traceback.format_stack()

  matplotlib.use('svg')  # faster

autoanchor: Analyzing anchors... anchors/target = 4.20, Best Possible Recall (BPR) = 1.0000
Aborted (core dumped)
(lpr_env) aiserver@aiserver-H110-D3:~/coding/GitHub/yolov5$ /home/aiserver/anaconda3/envs/lpr_env/lib/python3.7/multiprocessing/semaphore_tracker.py:144: UserWarning: semaphore_tracker: There appear to be 6 leaked semaphores to clean up at shutdown
  len(cache))

What am I supposed to do?

Thank you.

**Comment by glenn-jocher** [comment]
@leeyunhome your YOLOv5 code is very out of date, before looking at problems with it you should start with updating it.

**Comment by glenn-jocher** [comment]
@leeyunhome 👋 hi, thanks for letting us know about this problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to get started investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

* ✅ **Minimal** – Use as little code as possible that still produces the same problem
* ✅ **Complete** – Provide **all** parts someone else needs to reproduce your problem in the question itself
* ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

In addition to the above requirements, for [Ultralytics](https://ultralytics.com/) to provide assistance your code should be:

* ✅ **Current** – **Verify that your code is up-to-date with current GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been resolved by previous commits.**
* ✅ **Unmodified** – Your problem must be reproducible without any modifications to the codebase in this repository. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all of the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) and providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃
